pkg/server/k8s/client.go

```diff
@@ -312,16 +311,133 @@ func (k *k8sClient) Limits(namespace, name string) (*app.Limits, error) {
 	return lim, nil
 }
 
-func newInClusterK8sClient() (Client, error) {
-	conf, err := restclient.InClusterConfig()
+func (k *k8sClient) CreateDeploy(deploySpec *deploy.DeploySpec) error {
+	replicas := k.currentPodReplicasFromDeploy(deploySpec.Namespace, deploySpec.Name)
+	deployYaml := deploySpecToK8sDeploy(deploySpec, replicas)
+	_, err := k.kc.Deployments(deploySpec.Namespace).Update(deployYaml)
+	if k.IsNotFound(err) {
+		_, err = k.kc.Deployments(deploySpec.Namespace).Create(deployYaml)
+	}
+	return err
+}
+
+func (k *k8sClient) PodRun(podSpec *deploy.PodSpec) (io.ReadCloser, <-chan int, error) {
+	podYaml := podSpecToK8sPod(podSpec)
+	pod, err := k.kc.Pods(podSpec.Namespace).Create(podYaml)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	exitCodeChan := make(chan int)
+	r, w := io.Pipe()
+	go func() {
+		defer func() {
+			w.Close()
+			close(exitCodeChan)
+		}()
+
+		if err := k.waitPodStart(pod, 1*time.Second, 5*time.Minute); err != nil {
+			return
+		}
+
+		stream, err := k.PodLogs(podSpec.Namespace, podSpec.Name, 10, true)
+		if err != nil {
+			return
+		}
+		io.Copy(w, stream)
+
+		if err = k.waitPodEnd(pod, 1*time.Second, 5*time.Minute); err != nil {
```

Comment:
For deploy safety we need more than 5 minutes of finish timeout, maybe more args. Also we may increase the poll interval (less cpu intensive)